doc/topics/grains/index.rst

```diff
@@ -132,6 +132,15 @@ Using Jinja templating, only one match entry needs to be defined.
 Writing Grains
 ==============
 
+Consider adding Gains when the information is needed for targeting,
+the grains should only contain the **bare minimum data** required. Also think
+about other platforms/operating systems, and if the grain key name and data
+structure can be named and design to support many platforms, operating systems
+or applications. Most of the time execution module called from a JINJA template
```

Comment:
"Most of the time execution module" ?

Maybe "Most of the time, an execution module" would be better.